### PR TITLE
Add subpackage for RNN related code.

### PR DESCRIPTION
**Pull request by qlzh727**
Slowly bring over the tf.contrib.rnn code to addons.

The selection criteria so far are:
1. Previous requested by user.
2. From widely cited research paper (more than 100 based on Google scholar).

The initial commit only contains one rnn cell so far (NASCell), expect more cells to come.

### CODE REVIEW

**Comment by qlzh727** [comment]
Done. Thanks for the review.

File: tensorflow_addons/rnn/README.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ * Inherit from either `keras.layers.AbstractRNNCell` or `keras.layers.Layer` with
+   required properties.
+ * [Register as a Keras global object](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/utils/python/keras_utils.py)
+  so it can be serialized properly.
+ * Add the addon to the `py_library` in this sub-package's BUILD file.
+
+#### Testing Requirements
+ * When applicable, run all tests with TensorFlow's
+   `@run_in_graph_and_eager_modes` (for test method)
+   or `run_all_in_graph_and_eager_modes` (for TestCase subclass)
```
Comment (seanpmorgan) [inline tensorflow_addons/rnn/README.md]: nit: can you add the decorator `@` symbol

Comment (qlzh727) [reply tensorflow_addons/rnn/README.md]: Done.

**Comment by qlzh727** [comment]
Done. Updated doc and gitowner file.

**Pull request merged**